### PR TITLE
Docs: Fix links to internal files

### PR DESCRIPTION
**Pull request by manuzhang**

### CODE REVIEW

**Comment by manuzhang** [comment]
@bitsondatadev PTLK when you return from vacation.

**Review by bitsondatadev — CHANGES_REQUESTED** [review]
Some links need to be made relative. Otherwise this look good. Thanks @manuzhang

File: docs/docs/configuration.md [hunk 1]
```diff
@@ -122,16 +122,16 @@ The value of these properties are not persisted as a part of the table metadata.
...
-| io-impl                           | null               | a custom `FileIO` implementation to use in a catalog   |
-| warehouse                         | null               | the root path of the data warehouse                    |
-| uri                               | null               | a URI string, such as Hive metastore URI               |
-| clients                           | 2                  | client pool size                                       |
-| cache-enabled                     | true               | Whether to cache catalog entries |
-| cache.expiration-interval-ms      | 30000              | How long catalog entries are locally cached, in milliseconds; 0 disables caching, negative values disable expiration |
-| metrics-reporter-impl | org.apache.iceberg.metrics.LoggingMetricsReporter | Custom `MetricsReporter` implementation to use in a catalog. See the [Metrics reporting](metrics-reporting) section for additional details |
+| Property                          | Default            | Description                                                                                                                                   |
```
Comment (nastra) [inline docs/docs/configuration.md:125]: unrelated whitespace changes

File: docs/docs/spark-configuration.md [hunk 2]
```diff
@@ -176,19 +176,19 @@ df.write
...
+| ---------------------- | -------------------------- |----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| write-format           | Table write.format.default | File format to use for this write operation; parquet, avro, or orc                                                                                                                                                                                                                                                               |
+| target-file-size-bytes | As per table property      | Overrides this table's write.target-file-size-bytes                                                                                                                                                                                                                                                                              |
```
Comment (nastra) [inline docs/docs/spark-configuration.md:182]: unrelated whitespace changes

File: format/spec.md [hunk 3]
```diff
@@ -689,20 +689,20 @@ For serialization details, see Appendix C.
...
 Statistics files metadata within `statistics` table metadata field is a struct with the following fields:
 
-| v1 | v2 | Field name | Type | Description |
-|----|----|------------|------|-------------|
-| _required_ | _required_ | **`snapshot-id`** | `string` | ID of the Iceberg table's snapshot the statistics file is associated with. |
-| _required_ | _required_ | **`statistics-path`** | `string` | Path of the statistics file. See [Puffin file format](../puffin-spec). |
-| _required_ | _required_ | **`file-size-in-bytes`** | `long` | Size of the statistics file. |
-| _required_ | _required_ | **`file-footer-size-in-bytes`** | `long` | Total size of the statistics file's footer (not the footer payload size). See [Puffin file format](../puffin-spec) for footer definition. |
+| v1 | v2 | Field name | Type | Description                                                                                                                               |
+|----|----|------------|------|-------------------------------------------------------------------------------------------------------------------------------------------|
```
Comment (nastra) [inline format/spec.md:699]: unrelated whitespace changes

File: docs/docs/configuration.md (hunk 1 above)
Comment (manuzhang) [reply docs/docs/configuration.md:125]: shall we align columns after line length is changed?

Comment (nastra) [reply docs/docs/configuration.md:125]: it will be displayed correctly on the website, but it adds lots of visual noise for reviewing, so I'd not adjust the whitespace

**Comment by manuzhang** [comment]
@nastra @bitsondatadev Can you please check this again?

File: docs/docs/flink.md [hunk 4]
```diff
@@ -24,20 +24,20 @@ Apache Iceberg supports both [Apache Flink](https://flink.apache.org/)'s DataStr
...
-| [DataStream read](#reading-with-datastream)                 | ✔️ ️  |                                                                                        |
-| [DataStream append](#appending-data)                        | ✔️ ️  |                                                                                        |
-| [DataStream overwrite](#overwrite-data)                     | ✔️ ️  |                                                                                        |
-| [Metadata tables](#inspecting-tables)                       | ✔️    |                                                                                        |
-| [Rewrite files action](#rewrite-files-action)               | ✔️ ️  |                                                                                        |
+| [SQL create catalog](../flink-ddl.md#create-catalog) | ✔️    |                                                                                        |
+| [SQL create database](../flink-ddl.md#create-database)                     | ✔️    |                                                                                        |
```
Comment (nastra) [inline docs/docs/flink.md:28]:
```suggestion
| [SQL create database](../flink-ddl.md#create-database) | ✔️    |                                                                                        |
```

File: format/spec.md [hunk 5]
```diff
@@ -60,7 +60,7 @@ In addition to row-level deletes, version 2 makes some requirements stricter for
 
 ## Overview
 
-![Iceberg snapshot structure](../../../img/iceberg-metadata.png)
+![Iceberg snapshot structure](assets/images/iceberg-metadata.png)
```
Comment (nastra) [inline format/spec.md:63]: this was just fixed, can you please rebase?

File: docs/docs/flink.md (hunk 4 above)
Comment (manuzhang) [reply docs/docs/flink.md:28]: This will create whitespace change. Don't we want to avoid it?

Comment (nastra) [reply docs/docs/flink.md:28]: yes, but the formatting of the table looks already weird

File: docs/docs/flink.md [hunk 6]
```diff
@@ -24,20 +24,20 @@ Apache Iceberg supports both [Apache Flink](https://flink.apache.org/)'s DataStr
...
-| [SQL insert overwrite](#insert-overwrite)                   | ✔️ ️  |                                                                                        |
-| [DataStream read](#reading-with-datastream)                 | ✔️ ️  |                                                                                        |
-| [DataStream append](#appending-data)                        | ✔️ ️  |                                                                                        |
-| [DataStream overwrite](#overwrite-data)                     | ✔️ ️  |                                                                                        |
-| [Metadata tables](#inspecting-tables)                       | ✔️    |                                                                                        |
-| [Rewrite files action](#rewrite-files-action)               | ✔️ ️  |                                                                                        |
+| [SQL create catalog](../flink-ddl.md#create-catalog) | ✔️    |                                                                                        |
```
Comment (nastra) [inline docs/docs/flink.md:27]: it seems only this row was updated with the whitespacing changes but not the others

Comment (manuzhang) [reply docs/docs/flink.md:27]: are you referring to a previous change? In this PR, this line has no whitespace change

Comment (nastra) [reply docs/docs/flink.md:27]: nvm. Formatting these tables is quite annoying whenever some text changes, so let's just go with the diff you have

**Review by nastra — APPROVED** [review]
